### PR TITLE
Eliminate some occurrences of ts-ignore

### PR DESCRIPTION
**Pull request by willeastcott**
The viewer sourcebase has 58 occurrences of `ts-ignore`. This PR eliminates 22 of them leaving just 36 more to fix. 

This PR also:

* Updates some PlayCanvas function calls to conform with the currently supported API.
* Updates some function expressions  to be arrow functions.
* Fixes some spelling mistakes.

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: src/viewer.ts [hunk 1]
```diff
@@ -392,15 +391,18 @@ class Viewer {
...
                 addressV: pc.ADDRESS_CLAMP_TO_EDGE,
                 fixCubemapSeams: true,
                 mipmaps: false
             });
-            // @ts-ignore
-            pc.reprojectTexture(device, cubemaps[1] || skybox, prefilter, specPower[i], 4096);
+
+            pc.reprojectTexture(cubemaps[1] || skybox, prefilter, {
+                numSamples: 4096,
+                specularPower: specPower[i] === undefined ? 1 : specPower[i]
```
Comment (willeastcott) [inline src/viewer.ts:403]: This looks weird to me @slimbuck. How come the first element in the `specPower` array is `undefined`?

**Comment by kungfooman** [comment]
Instead of closing my PR, this is improving into the same direction, while my PR has a focus on other `ts-ignore`'s like:

```ts
// @ts-ignore TODO anim property missing from pc.Entity
entity.anim.loadStateGraph(new pc.AnimStateGraph({
```

E.g. the reason you couldn't remove that one is because the `playcanvas.d.ts` still lacks all the crucial type information:

image

Which is what my PlayCanvas PR is about: https://github.com/playcanvas/engine/pull/2986

Until you fix that, you can't fix this. And that's not merged because no one cares to update the JSDoc generation to inline JSDoc `typedef`'s.

File: src/viewer.ts [hunk 2]
```diff
@@ -879,14 +872,17 @@ class Viewer {
     }
 
     setTonemapping(tonemapping: string) {
-        const mapping = {
+        type mapping = {
+            [key: string]: number
+        };
+        const mappingLookup: mapping = {
```
Comment (kungfooman) [inline src/viewer.ts:878]:
```suggestion
        const mappingLookup: Record<string, number> = {
```

I would remove the entire `mappingLookup` renaming to make it shorter again, but GitHub doesn't allow code suggestions for deleted lines.

Comment (willeastcott) [reply src/viewer.ts]: Ah cool suggestion, thanks, that's much better! 😄

**Comment by willeastcott** [comment]
@kungfooman I'm starting to add issues to the Engine repo that cover the remaining `ts-ignore` directives:

* https://github.com/playcanvas/engine/issues/3673
* https://github.com/playcanvas/engine/issues/3674

The `AnimStateGraph` constructor is highlighted in the second one. I think it's oversimplifying to say that no-one cares about updating the JSDoc generation to inline JSDoc `typedef`s. The PR you made on the engine repo has outstanding problems - as far as I can tell in terms of the API reference that's generated. And while we want to support Typescript development, it feels like there's a balance between that and simplicity/maintainability of our JSDoc comments. That PR adds ~100 lines of JSDoc to define one constructor argument. (Another issue I have is that the types have underscores which is not done anywhere else in the codebase, but obviously that's not a hard thing to change). But those JSDoc blocks are hard to read and swamps the file (just my opinion). For now, I would rather just say `data` is an `object` and document it in a nice and readable way.

File: src/viewer.ts (hunk 1 above)
Comment (slimbuck) [reply src/viewer.ts:403]: Yup, thanks.

Comment (willeastcott) [reply src/viewer.ts]: Done. 👍

**Review by slimbuck — APPROVED** [review]
this is awsome

**Pull request merged**